### PR TITLE
Memory searcher cleanups

### PR DESCRIPTION
**Pull request by leculver**
- Add todo marker where we need to improve memory searching in the future.
- Remove Console.WriteLines.
- Change SearchMemory to use a length.

### CODE REVIEW

File: src/Microsoft.Diagnostics.Runtime.Tests/src/MemoryReaderTests.cs [hunk 1]
```diff
@@ -43,11 +43,11 @@ public void SearchMemoryTest()
                         if (!dt.DataReader.Read(expectedOffset, slice, out int read) || read != slice.Length)
                             continue;
 
-                        ulong addressFound = dt.DataReader.SearchMemory(segment.Start, segment.End, buffer.Slice(0, i + sizeof(ulong)));
+                        ulong addressFound = dt.DataReader.SearchMemory(segment.Start, (int)segment.Length, buffer.Slice(0, i + sizeof(ulong)));
 
                         // There could still accidently be a pattern that matches this somewhere
                         while (addressFound != 0 && addressFound < expectedOffset)
-                            addressFound = dt.DataReader.SearchMemory(addressFound + 1, segment.End, buffer.Slice(0, i + sizeof(ulong)));
+                            addressFound = dt.DataReader.SearchMemory(addressFound + 1, (int)segment.End, buffer.Slice(0, i + sizeof(ulong)));
```
Comment (NextTurn) [inline src/Microsoft.Diagnostics.Runtime.Tests/src/MemoryReaderTests.cs]: Arithmetic is needed here.

Well, from the uses of that method it does seem that accepting an end address is more natural... I have no idea why most APIs in the core library prefer a length though.

Comment (leculver) [reply src/Microsoft.Diagnostics.Runtime.Tests/src/MemoryReaderTests.cs]: It's surprising that the test passed.  I must have gotten lucky with that underflow.

Comment (leculver) [reply src/Microsoft.Diagnostics.Runtime.Tests/src/MemoryReaderTests.cs]: Taking a length makes a lot of sense when working with strings, arrays, and lists.  I'm happy to follow that convention here.  I just need more coffee to stop failing at basic coding.  =)

File: src/Microsoft.Diagnostics.Runtime/src/Utilities/Platform/LinuxFunctions.cs [hunk 2]
```diff
@@ -231,6 +232,8 @@ internal override unsafe bool GetFileVersion(string dll, out int major, out int
 
                 address += s_versionLength;
 
+                // TODO:  This should be cleaned up to not read byte by byte in the future.  Leaving it here
```
Comment (NextTurn) [inline src/Microsoft.Diagnostics.Runtime/src/Utilities/Platform/LinuxFunctions.cs]: Oh that's a `StreamAddressSpace` wrapping a `FileStream`. Just use the `FileStream` with `Seek` and `Read` instead and eliminate the byte-to-byte reads.

**Pull request merged**